src/Nethermind/Nethermind.Mev/Data/MevBundle.cs

```diff
@@ -51,15 +61,37 @@ public MevBundle(IReadOnlyList<Transaction> transactions, long blockNumber, UInt
         public UInt256 MaxTimestamp { get; }
         
         public UInt256 MinTimestamp { get; }
+        
+        public Keccak Hash { get; }
+
+        public int PoolIndex { get; }
 
         public bool Equals(MevBundle? other)
         {
             if (ReferenceEquals(null, other)) return false;
             if (ReferenceEquals(this, other)) return true;
-            return Transactions.Select(t => t.Hash).SequenceEqual(other.Transactions.Select(t => t.Hash))
-                   && BlockNumber == other.BlockNumber
-                   && MaxTimestamp.Equals(other.MaxTimestamp)
-                   && MinTimestamp.Equals(other.MinTimestamp);
+            return Equals(Hash, other.Hash);
+        }
+
+        public int CompareTo(MevBundle? other)
```

Comment:
this should not be a responsibility of this class, move to a separate comparer class